src/algopy_testing/op.py

```diff
@@ -938,69 +1000,315 @@ def _get_asset_holding(
+    ) -> tuple[algopy.UInt64, bool]:
+        return self._get_app_param_from_ctx(a, "local_num_uint")
+
+    def app_local_num_byte_slice(
+        self, a: algopy.Application | algopy.UInt64 | int, /
+    ) -> tuple[algopy.UInt64, bool]:
+        return self._get_app_param_from_ctx(a, "local_num_byte_slice")
+
+    def app_extra_program_pages(
+        self, a: algopy.Application | algopy.UInt64 | int, /
+    ) -> tuple[algopy.UInt64, bool]:
+        return self._get_app_param_from_ctx(a, "extra_program_pages")
+
+    def app_creator(
+        self, a: algopy.Application | algopy.UInt64 | int, /
+    ) -> tuple[algopy.Account, bool]:
+        return self._get_app_param_from_ctx(a, "creator")
+
+    def app_address(
+        self, a: algopy.Application | algopy.UInt64 | int, /
+    ) -> tuple[algopy.Account, bool]:
+        return self._get_app_param_from_ctx(a, "address")
 
 
 AppParamsGet = _AppParamsGet()
 
 
 class _AppLocal:
-    def __getattr__(self, name: str) -> Any:
-        raise NotImplementedError(
-            f"AppLocal.{name} is currently not available as a native "
-            "`algorand-python-testing` type. Use your own preferred testing "
-            "framework of choice to mock the behaviour."
+    def _get_local_state(self, key: bytes) -> algopy.LocalState[Any]:
+        from algopy_testing import get_test_context
+
+        test_context = get_test_context()
+        if not test_context or not test_context._active_contract:
+            raise ValueError("No active contract or test context found.")
+
+        local_states = test_context._active_contract._get_local_states()
+        local_state = local_states.get(key)
+        if local_state is None:
+            raise ValueError(f"Local state with key {key.decode()} not found")
+        return local_state
+
+    def _get_key(self, b: algopy.Bytes | bytes) -> bytes:
+        import algopy
+
+        return b.value if isinstance(b, algopy.Bytes) else b
+
+    def _get_contract(
+        self,
+        b: algopy.Application | algopy.UInt64 | int,
+    ) -> algopy.Contract | algopy.ARC4Contract:
+        import algopy
+
+        from algopy_testing import get_test_context
+
+        test_context = get_test_context()
+        if not test_context:
+            raise ValueError("Test context is not initialized!")
+
+        app_id = int(b.id) if isinstance(b, algopy.Application) else int(b)
+        contract = test_context._app_id_to_contract.get(app_id)
+        if contract is None:
+            raise ValueError(f"Contract with app id {b} not found")
+        return contract
+
+    def get_bytes(
+        self, a: algopy.Account | algopy.UInt64 | int, b: algopy.Bytes | bytes, /
+    ) -> algopy.Bytes:
+        import algopy
+
+        try:
+            local_state = self._get_local_state(self._get_key(b)).get(a, 0)
+            return algopy.Bytes(local_state) if local_state else algopy.UInt64(0)  # type: ignore[return-value]
```

Comment:
this won't be correct if the local state values is `b""` should use `None` as the default and then test if `local_state is not None` to determine whether to return `UInt64(0)`

Reply:
Another things to consider is that the `LocalState` proxy will could be holding one of the non primitive types e.g. an arc4 value or an `Account`  etc., but this low level op just returns the underlying Bytes. 

We will need to convert the value obtained from LocalState into a `Bytes`. 
This is a case where having the global/local state seperate from the contract and the state proxies might be worth considering.

Reply:
added arc4 parsers to all get byte and get ex bytes methods in local and global classes